test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1574,13 +1574,29 @@ public void indexRandom(boolean forceRefresh, boolean dummyDocuments, boolean ma
         }
     }
 
+    /*
+     * This method deletes the given bogus documents
+     * @param bogusIds doc ids to be deleted
+     */
+    protected void deleteBogusDocs(Set<List<String>> bogusIds) {
+        for (List<String> doc : bogusIds) {
+            assertEquals(
+                "failed to delete a dummy doc [" + doc.get(0) + "][" + doc.get(1) + "]",
+                DocWriteResponse.Result.DELETED,
+                client().prepareDelete(doc.get(0), doc.get(1)).setRouting(doc.get(1)).get().getResult()
+            );
+        }
+    }
+
     /*
      * This method ingests bogus documents for the given indices such that multiple slices
      * are formed. This is useful for testing with the concurrent search use-case as it creates
-     * multiple slices based on segment count.
-     * @param indices         the indices in which bogus documents should be ingested
+     * multiple slices based on segment count. When calling this method with value of toDeleteBogusDocs
+     * as true, don't forget to call deleteBogusDocs at the end of the test to delete the bogus docs
+     * @param toDeleteBogusDocs  provides an option to delete the bogus docs optionally
+     * @param indices            the indices in which bogus documents should be ingested
      * */
-    protected void indexRandomForMultipleSlices(String... indices) throws InterruptedException {
+    protected Set<List<String>> indexRandomForMultipleSlices(boolean toDeleteBogusDocs, String... indices) throws InterruptedException {
```

Comment:
If there are multiple indices how do we map the `bogusIds` to a given index as `Set` is an unordered collection?

Reply:
the map of an index to bogus id doesnt matter, as bogus ids set would be returned for all the indices passed in the function call and deletion will be done for the same set of bogus ids (which will be bogus doc for all the indices)